### ISSUE TITLE
The Strategy of facial expression recogition 

### ISSUE THREAD

**Issue by zhanglaplace** [oarriaga/face_classification#7]
Hi , It's a pleasure to see such good open source code。
in face, i am also doing the research of facial expression recognition .
i want to know what  effective skills have you taken.
i see the CNN structure is ordinary except depth.
and what's the effect when the real test is done.

**Comment by oarriaga** [comment]
Hello @zhanglaplace,
thank you for the interest in the code, I tried different architectures trying to create the smallest model with the highest accuracy, mostly inspired by the very well explained "size principle/Occam’s razor" and "strong sampling assumption" concepts in the ML book by Kevin Murphy.

The two most important things that I found out about the models were:
* The kernel size of the first convolution layer has to be bigger than what I expected (7 x 7).
* We can train a model that does not require a fully connected layer at the end (which can contain a considerable amount of parameters) but instead we can use the global average pooling "trick" presented in the Inception architecture.

Having said this there are things that I have not tested myself such as:
* How much does batch normalization help?
* How much does dropout help?
* Can we reduce the number of feature maps in each layer?

Finally, with a very qualitative interpretation,  I believe the model that performs better "in-the-wild" is not the one with the highest accuracy.  Therefore I provided another pre-trained model for the emotions.

**Comment by zhanglaplace** [comment]
Thank you for your answer. the image in fer2013 datasets is 48*48 pixels  , it's too small , so why the first convolution layers should be bigger . and have you fine-tune the model on any others datasets ? sorry. my english is poor.

**Comment by oarriaga** [comment]
Yes, exactly the size is 48*48 the initial kernel size is big in comparison to the image size. Initially a big kernel size was making the model converge. However, I am currently training new models with a smaller initial kernel size and it seems the models are convering as well. We could plot the kernels and the feature maps of the first set of convolutions to understand what are the kernels filtering. I have not pre-trained on any other network. Another idea could be to do transfer learning with the pre-trained weigths from the gender classification to train the emotion classifier and see if there is any improvement.

**Comment by zhanglaplace** [comment]
@oarriaga yeah，Facial expression recognition is mainly the lack of relatively large data sets and data pictures，if the training image enough large, i think the accuracy  mush improve.
